### ISSUE TITLE
CORS options openid-configuration

### ISSUE THREAD

**Issue by jensluch** [panva/node-oidc-provider#438]
**Describe the bug**
After 5.5.3 -> 5.5.4 changed headers returned by
HTTP OPTIONS 
```/.well-known/openid-configuration```

Was:
```
Access-Control-Allow-Methods: GET,HEAD,PUT,POST,DELETE,PATCH
```
Now:
```
Allow: OPTIONS,ANY,GET,HEAD,POST
```
It not looks like minor changes because some CORS issues may happen.

**To Reproduce**

Try CORS HTTP OPTIONS ```/.well-known/openid-configuration``` request

**Comment by panva** [comment]
@jensluch thanks for reporting this, at least i'll have to write tests for default cors setup :)

**Comment by panva** [comment]
Two issues at hand that got fixed now in `v5.5.5`

1) the new router was blocking preflight middleware from running. see https://github.com/koajs/trie-router/issues/40
2) the `allowMethods` option for https://github.com/koajs/cors was passed as `allowedMethods` - forever mind you

As a result 1) preflight is now fixed 2) is actually returning the correct methods

@jensluch thanks for bringing it up, i'll take a stab at the tests later
